src/vma/sock/sockinfo.cpp

```diff
@@ -350,6 +405,25 @@ int sockinfo::setsockopt(int __level, int __optname, const void *__optval, sockl
 				si_logdbg("SOL_SOCKET, %s=\"???\" - NOT HANDLED, optval == NULL", setsockopt_so_opt_to_str(__optname));
 			}
 			break;
+		case SO_VMA_RING_ALLOC_LOGIC:
+			if (__optval) {
+				if (__optlen == sizeof(vma_ring_alloc_logic_attr)) {
+					vma_ring_alloc_logic_attr *attr = (vma_ring_alloc_logic_attr *)__optval;
+					return set_ring_attr(attr);
+				}
+				else {
+					ret = SOCKOPT_NO_VMA_SUPPORT;
+					si_logdbg("SOL_SOCKET, %s=\"???\" - bad length expected %d got %d",
+						  setsockopt_so_opt_to_str(__optname),
+						  sizeof(vma_ring_alloc_logic_attr), __optlen);
+					break;
+				}
+			}
+			else {
+				ret = SOCKOPT_NO_VMA_SUPPORT;
+				si_logdbg("SOL_SOCKET, %s=\"???\" - NOT HANDLED, optval == NULL", setsockopt_so_opt_to_str(__optname));
+			}
+			break;
```

Comment:
Do we want to add support for getsockopt  ?

Reply:
Added